$NetBSD: patch-ak,v 1.1.2.2 2012/11/15 15:50:10 tron Exp $

Avoid a duplicate case statement under powerpc, as the FFI_TYPE_LONGDOUBLE
and FFI_TYPE_DOUBLE constants are defined to the same value.

--- src/powerpc/ffi.c.orig	2012-04-12 02:46:06.000000000 +0000
+++ src/powerpc/ffi.c
@@ -664,9 +664,11 @@ ffi_prep_cif_machdep (ffi_cif *cif)
   switch (type)
     {
 #ifndef __NO_FPRS__
+#if FFI_TYPE_LONGDOUBLE != FFI_TYPE_DOUBLE
     case FFI_TYPE_LONGDOUBLE:
       flags |= FLAG_RETURNS_128BITS;
       /* Fall through.  */
+#endif
     case FFI_TYPE_DOUBLE:
       flags |= FLAG_RETURNS_64BITS;
       /* Fall through.  */
